lib/src/signed_video_h26x_auth.c

```diff
@@ -102,21 +102,19 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
  * Detects if there are any missing SEI messages based on the GOP counter and updates the GOP state.
  */
 static void
-detect_lost_sei_and_update_gop_state(signed_video_t *self)
+detect_lost_sei(signed_video_t *self)
 {
-  // Get the last GOP counter before updating.
+  // Get the last GOP counter.
   uint32_t exp_gop_number = self->gop_info->latest_validated_gop + 1;
   // Compare new with last number of GOPs to detect potentially lost SEIs.
   uint32_t new_gop_number = self->gop_info->global_gop_counter;
   int64_t potentially_missed_gops = (int64_t)new_gop_number - exp_gop_number;
-  // If number of |potentially_missed_gops| is negative, we have either lost SEIs together with a
-  // wraparound of |global_gop_counter|, or a reset of Signed Video was done on the camera. The
-  // correct number of lost SEIs is of less importance, since we only want to know If we have lost
-  // any. Therefore, make sure we map the value into the positive side only. It is possible to
-  // signal to the validation side that a reset was done on the camera, but it is still not
-  // possible to validate pending NALUs.
-  if (potentially_missed_gops < 0 && new_gop_number >= exp_gop_number)
-    potentially_missed_gops += INT64_MAX;
+  // To estimate whether a wraparound has occurred, we check if the adjusted value
+  // is within a specific range that indicates a likely wraparound. If so, we adjust
+  // the value accordingly. This approach cannot definitively differentiate between
+  // a reset and a wraparound but provides a practical heuristic for handling the situation.
+  bool is_wraparound = (potentially_missed_gops + INT64_MAX) < (INT64_MAX / 2);
+  if (is_wraparound) potentially_missed_gops = +INT64_MAX;
```

Comment:
This looks wrong. Shouldn't it be `potentionally_missed_gops += INT64`
Also, make a line break after the if.

Now I'm going to make life miserable for you. What happens if someone have interchanged two SEI frames? That belongs to a separate PR, though.

Reply:
There is no macro defined as INT64  theri is only INT64_MAX